common/src/Microsoft.Azure.IIoT.Agent.Framework/src/Storage/Database/Services/WorkerDatabase.cs

```diff
@@ -27,6 +27,7 @@ public class WorkerDatabase : IWorkerRegistry, IWorkerRepository {
             var database = databaseServer.OpenAsync(databaseRegistryConfig.DatabaseName).Result;
             var container = database.OpenContainerAsync(databaseRegistryConfig.ContainerName).Result;
             _documents = container.AsDocuments();
+            _lock = new SemaphoreSlim(1, 1);
```

Comment:
Can we have a lock per `WorkerId` so that operations for different workers can execute in parallel ?

Reply:
Not really, the intent here was to guard the access to _documents collection itself. I think the etag mismatch that causes the issue happens exactly when accessing the different workers in paralel.